### PR TITLE
Update Docker to ROCm 4.5 and support Navi on Jenkins

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -110,6 +112,11 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
         def debug_flags = "-g -O2 -fno-omit-frame-pointer -fsanitize=${sanitizers} -fno-sanitize-recover=${sanitizers}"
         cmake_build("/opt/rocm/llvm/bin/clang++", "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_GPU=Off -DMIGRAPHX_ENABLE_CPU=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}'")
     }
+}, clang_release_navi: rocmnode('navi21') { cmake_build ->
+    stage('HIP Clang Release Navi') {
+        cmake_build("/opt/rocm/llvm/bin/clang++", "-DCMAKE_BUILD_TYPE=release")
+        stash includes: 'build/*.deb', name: 'migraphx-package'
```
Comment (pfultz2) [inline Jenkinsfile:118]: Dont stash the debs. Onnx runtime only runs on vega, right now.

File: test/msgpack.cpp [hunk 2]
```diff
@@ -11,7 +11,7 @@ std::vector<char> msgpack_buffer(const T& src)
     msgpack::pack(buffer, src);
     buffer.seekg(0);
     std::string str = buffer.str();
-    return std::vector<char>(str.data(), str.data() + str.size());
+    return std::vector<char>(str.data(), str.data() + str.size()); // NOLINT
```
Comment (pfultz2) [inline test/msgpack.cpp:14]: Why does this use `NOLINT`?

Comment (kahmed10) [reply test/msgpack.cpp:14]: I tried rewriting using a brace initializer list, but then I get a tidy error that the local variable `str` should not be used within the initializer list when returning from the function. I had a workaround commit where I just replaced `str` with `buffer.str()`, but then the msgpack test failed.

File: Jenkinsfile (hunk 1 above)
Comment (kahmed10) [reply Jenkinsfile]: fixed

**Review by umangyadav — APPROVED** [review]

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**